src/ORM/ArrayList.php

```diff
@@ -127,7 +127,7 @@ public function exists()
     public function getIterator(): Traversable
     {
         foreach ($this->items as $i => $item) {
-            if (is_array($item)) {
+            if (is_array($item) && !array_is_list($item)) {
```

Comment:
Ultimately the way `ArrayList` works needs to be re-evaluated at some stage anyway - only converting in the iterator but not when fetching items directly via `find()` and other methods leads to unexpected results. So I think this is fine for now even though it allows a setup that used to cause an explicit error.